File: Code/Websites/DanpheEMR/wwwroot/DanpheApp/src/app/incentive/shared/incentive-transaction-items-vm.ts
import { FormGroup, FormBuilder, Validators } from '@angular/forms';

export class IncentiveTransactionItemsVM {

  public TransactionDate: string = null;
  public FiscalYearId: number;
  public InvoiceNo: string = null;
  public PriceCategory: string = null;
  public BillItemPriceId: number = null;
  public BillingTransactionId: number = null;
  public BillingTransactionItemId: number = null;
  public PatientId: number = null;
  public PatientName: string = null;
  public PatientCode: string = null;
  public ItemName: string = null;
  public TotalAmount: number = null;
  public ReferredByEmpId: number = null;
  public ReferredByEmpName: string = null;
  public ReferredByPercent: number = null;
  public ReferralAmount: number = null;
  public AssignedToEmpId: number = null;
  public AssignedToEmpName: string = null;
  public AssignedToPercent: number = null;
  public AssignedToAmount: number = null;
  public IsSelected: boolean = false;//only for client side.
}

File: Code/Components/DanpheEMR.ServerModel/IncentiveModels/IncentiveTransactionItemsVM.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DanpheEMR.ServerModel.IncentiveModels
{
    class IncentiveTransactionItemsVM
    {
        public DateTime TransactionDate { get; set; }
        public int FiscalYearId { get; set; }
        public string InvoiceNo { get; set; }
        public string PriceCategory { get; set; }
        public int BillingTransactionItemId { get; set; }
        public int PatientId { get; set; }
        public string PatientName { get; set; }
        public string PatientCode { get; set; }
        public string ItemName { get; set; }
        public float? TotalAmount { get; set; }
        public int? ReferredByEmpId { get; set; }
        public string ReferredByEmpName { get; set; }
        public float? ReferredByPercent { get; set; }
        public float? ReferralAmount { get; set; }
        public int? AssignedToEmpId { get; set; }
        public string AssignedToEmpName { get; set; }
        public float? AssignedToPercent { get; set; }
        public float? AssignedToAmount { get; set; }

    }
}
